### DISCUSSION TITLE
Query rule works only on page 1 of search results

### DISCUSSION THREAD

**Discussion by rikehuman** [microsoft-search/pnp-modern-search#2106] [Q&A] [answered]
My query rule fires as expected and the search results webpart shows the results on the **first** page. But selecting the next page (e.g. page2 or the last page) the search results webpart ignores the query rule - the _queryModification_ gets lost (v 4.6.1)

From Debug layout see the following on page 1 and page 2

**Page 1**

"filters": [],
    "**queryModification**": "52054 -ContentClass=urn:content-class:SPSPeople",
    "secondaryResults": [],
    "promotedResults": [
      {
        "title": "test2",
        "url": "https://www.google.com/",
        "description": "lorem ipsum"
      }
    ]
  },
  "paging": {
    "currentPageNumber": **1**
  },

****
**Page 2**

"filters": [],
    "**queryModification**": " microview ",  
    "secondaryResults": []
  },
  "paging": {
    "currentPageNumber": **2**
  },

On page 2 queryModification is my original query _microview_ but there should be the query rule _52054 -ContentClass=urn:content-class:SPSPeople_

**Comment by wobba** [comment, marked as answer]
This seems to be an API issue and I can repro using API or classic search /_layouts/15/osssearchresults.aspx. Best course of action is to log this with support with repro via classic search.
